usagov/usagov-2021
# USAGOV-1899 Benefits Redirect Modal

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1899

## Description

This adds a modal block-type for CMS users to create modals for the website.
It includes javascript to check the url for a modal query parameter and display a modal if one with a matching name exists.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



- Visit the Content blocks page `/admin/content/block`
- Click `+ Add content block`
- Select `Modal`
- Fill in the fields for the modal.
   - `Modal ID` will be used in the URL query parameter so keep it URL safe
   - `Button Text` can be anything since this button remains hidden
   - `Yes Button Text` will be displayed as the button to exit the modal
   - `No Button Text` should be left blank
   - Make sure you select the correct language for the banner.
- Click `Save`
- Visit any page of the site and add "?modal=`modal_id`" to end of the URL
- [ ] Confirm that the modal is displayed properly at any screen size
- [ ] Confirm that the URL parameter is automatically removed from the URL
- [ ] Confirm that you can close the modal
- [ ] Confirm that the modal does not reappear when navigating the site (Even with the browser's back button)
- [ ] Confirm that a spanish modal will only appear on spanish pages and english only on english pages

### Change Requirements



### Validation Steps

- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
